fox-tech/creds-fetcher
# Feature/CSPPIT-107/Implement AWS library

## Description
Implement library to handle AWS-STS authentication and saving credentials to file. 

## Ticket
[CSPPIT-107](https://foxjira.praecipio.com/browse/CSPPIT-107)

## Usage
````
a := aws.New(aws.Profile{})
a.GenerateCredentials(assertion)

````
## How to test this ticket
run `go test -race -count=1 -coverprofile=coverage.txt -covermode=atomic ./...` and all tests should pass.

For a end-to-end testing, this ticket requires [CSPPIT-106](https://foxjira.praecipio.com/browse/CSPPIT-106) and [CSPPIT-108](https://foxjira.praecipio.com/browse/CSPPIT-108)

For a rough approach, a temp main function can be created like so:
````

func main() {
	a := aws.New(aws.Profile{
		Name:         "poc-foxnews",
		RoleARN:      "", // aws role arn
		PrincipalARN: "", // aws provider arn
	})
	a.GenerateCredentials(assertion)
}
````

where `assertion` is a valid SAML token assertion. After, the following command should work:
`aws s3api list-buckets --profile poc-foxnews`
